File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse1.java
import java.nio.charset.StandardCharsets;

public class Lse1 {
    // Copies n bytes from src to dest; first arg dest, second src, third n
    public static void memCopy(byte[] dest, byte[] src, int n) {
        if (dest == null || src == null) throw new IllegalArgumentException("Null array");
        if (n < 0) throw new IllegalArgumentException("Negative length");
        if (n > dest.length || n > src.length) throw new IllegalArgumentException("Length exceeds array bounds");
        for (int i = 0; i < n; i++) {
            dest[i] = src[i];
        }
    }

    private static String hex(byte[] a, int n) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < n; i++) {
            sb.append(String.format("%02X", a[i] & 0xFF));
            if (i + 1 < n) sb.append(" ");
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        // Test 1: Copy full ASCII buffer
        byte[] src1 = "HelloWorld".getBytes(StandardCharsets.UTF_8);
        byte[] dest1 = new byte[10];
        memCopy(dest1, src1, 10);
        System.out.println("T1 HEX: " + hex(dest1, 10));
        System.out.println("T1 STR: " + new String(dest1, StandardCharsets.UTF_8));

        // Test 2: Partial copy
        byte[] src2 = "OpenAI".getBytes(StandardCharsets.UTF_8);
        byte[] dest2 = new byte[4];
        memCopy(dest2, src2, 4); // "Open"
        System.out.println("T2 HEX: " + hex(dest2, 4));
        System.out.println("T2 STR: " + new String(dest2, StandardCharsets.UTF_8));

        // Test 3: Zero-length copy (dest remains unchanged)
        byte[] src3 = "XYZ".getBytes(StandardCharsets.UTF_8);
        byte[] dest3 = new byte[] {(byte)0xEE, (byte)0xEE, (byte)0xEE};
        memCopy(dest3, src3, 0);
        System.out.println("T3 HEX: " + hex(dest3, dest3.length));

        // Test 4: Binary data copy
        byte[] src4 = new byte[] {0x00, 0x01, (byte)0xAB, (byte)0xFF, 0x10};
        byte[] dest4 = new byte[5];
        memCopy(dest4, src4, 5);
        System.out.println("T4 HEX: " + hex(dest4, 5));

        // Test 5: Copy first 7 bytes
        byte[] src5 = "1234567890".getBytes(StandardCharsets.UTF_8);
        byte[] dest5 = new byte[7];
        memCopy(dest5, src5, 7); // "1234567"
        System.out.println("T5 HEX: " + hex(dest5, 7));
        System.out.println("T5 STR: " + new String(dest5, StandardCharsets.UTF_8));
    }
}

File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse1.py
def mem_copy(dest: bytearray, src: bytes, n: int) -> None:
    if dest is None or src is None:
        raise ValueError("Null buffer")
    if n < 0:
        raise ValueError("Negative length")
    if n > len(dest) or n > len(src):
        raise ValueError("Length exceeds buffer bounds")
    for i in range(n):
        dest[i] = src[i]

def _hex(b: bytes, n: int) -> str:
    return " ".join(f"{b[i]:02X}" for i in range(n))

if __name__ == "__main__":
    # Test 1: Copy full ASCII buffer
    src1 = b"HelloWorld"
    dest1 = bytearray(10)
    mem_copy(dest1, src1, 10)
    print("T1 HEX:", _hex(dest1, 10))
    print("T1 STR:", dest1.decode("utf-8"))

    # Test 2: Partial copy
    src2 = b"OpenAI"
    dest2 = bytearray(4)
    mem_copy(dest2, src2, 4)  # "Open"
    print("T2 HEX:", _hex(dest2, 4))
    print("T2 STR:", dest2.decode("utf-8"))

    # Test 3: Zero-length copy (dest remains unchanged)
    src3 = b"XYZ"
    dest3 = bytearray([0xEE, 0xEE, 0xEE])
    mem_copy(dest3, src3, 0)
    print("T3 HEX:", _hex(dest3, len(dest3)))

    # Test 4: Binary data copy
    src4 = bytes([0x00, 0x01, 0xAB, 0xFF, 0x10])
    dest4 = bytearray(5)
    mem_copy(dest4, src4, 5)
    print("T4 HEX:", _hex(dest4, 5))

    # Test 5: Copy first 7 bytes
    src5 = b"1234567890"
    dest5 = bytearray(7)
    mem_copy(dest5, src5, 7)  # "1234567"
    print("T5 HEX:", _hex(dest5, 7))
    print("T5 STR:", dest5.decode("utf-8"))